### PR TITLE
[libc] Generate configure.rst from the JSON config information.

### PR DESCRIPTION
**Pull request by sivachandra**

### CODE REVIEW

File: libc/cmake/modules/LibcConfig.cmake [hunk 1]
```diff
@@ -135,3 +135,85 @@ function(load_libc_config config_file)
...
+    message(FATAL_ERROR "${config_file} does not exist")
+  endif()
+  file(READ ${config_file} json_config)
+  string(JSON group_count ERROR_VARIABLE json_error LENGTH ${json_config})
+  if(json_error)
+    message(FATAL_ERROR "${config_file}: ${json_error}")
+  endif()
+  if(${group_count} EQUAL 0)
+    message(FATAL_ERROR "${config_file}: Does not contain any config option groups")
+    return()
```
Comment (lntue) [inline libc/cmake/modules/LibcConfig.cmake:150]: Doesn't `FATAL_ERROR` from the line above stop cmake processing completely?

Comment (sivachandra) [reply libc/cmake/modules/LibcConfig.cmake:150]: Yes. The idea is that if there are no config option groups, there should be no `config.json` added at all.

Comment (lntue) [reply libc/cmake/modules/LibcConfig.cmake:150]: SGTM.  Then the `return()` in this line is not needed.

Comment (sivachandra) [reply libc/cmake/modules/LibcConfig.cmake]: Fixed now.

**Review by lntue — APPROVED** [review]

**Pull request merged**